scripts/index.js

```diff
@@ -1 +1,138 @@
-// add your solutions here
\ No newline at end of file
+// add your solutions here
+
+// Function Exercises
+
+//======POSITIVE NUMBERS===============//
+// -- Write a function which takes an array of numbers as input and returns a new array containing only the positive numbers in the given array.
+
+const numberList = [-1, 0, 2, -10, 7, -1];
+let positiveNumbers = [];
+
+numberList.forEach(function (originalNumbers) {
+    if (originalNumbers >= 0) {
+        positiveNumbers.push(originalNumbers);
+    }
+    return positiveNumbers;
+});
+//===========EVEN NUMBERS===================//
+let evenNumbers = [];
+numberList.forEach(function(originalNumbers) {
+    if (originalNumbers % 2 === 0) {
+        evenNumbers.push(originalNumbers);
+    }
+    return evenNumbers;
+});
+
+//==========SQUARE THE NUMBER==============//
+
+const squareNum = (num) => {
+    return num * num;
+};
+
+//output 1=1 2=4 3=9
+
+//=========CITITES 1=================//
+// Write a function which takes an array of city objects as input and returns a new array containg the cities whose temperature is cooler than 70 degrees
+
+var cities = [
+    { 
+        name: 'Los Angeles', 
+        temperature: 60.0
+    },
+    { 
+        name: 'Atlanta', 
+        temperature: 52.0 },
+    { 
+        name: 'Detroit', 
+        temperature: 48.0 
+    },
+    { 
+        name: 'New York', 
+        temperature: 80.0 
+    },
+];
+Object.values(cities);
+
+tempsOver70 = [];
+cities.forEach(function (temps) {
+    temps.temperature;
+    if (temps.temperature >= 70) {
+    tempsOver70.push(temps.name, temps.temperature);
+    }
+    return tempsOver70;
+    //console.log(temps.temperature)
+});
+
+//==========CITIES 2==============//
+// Write a function which takes an array of city objects like the above problem as input and returns an array of city names
+
+cityList = [];
+cities.forEach(function (names) {
+    names.name;
+    if (names.name){
+        cityList.push(names.name);
+    }
+    return cityList;
+});
+
+
+//=========GOOD JOB!==============//
+// Given an array of names print out 'Good Job, {{name}}!' for each person's name, one on a line
+
+const people = [
+    'Dom',
+    'Lyn',
+    'Kirk',
+    'Autumn',
+    'Trista',
+    'Jesslyn',
+    'Kevin',
+    'John',
+    'Eli',
+    'Juan',
+    'Robert',
+    'Keyur',
+    'Jason',
+    'Che',
+    'Ben'
+];
+
+people.forEach (function (firstName) {
+    console.log(`Good job, ${firstName}!`)
+});
+
+//==========3 TIMES==================// 
+
+fun = "Hello, world!"
+function call3Times(fun) {
+    console.log(fun);
+    console.log(fun);
+    console.log(fun);
+}
+
+call3Times(fun);
+
+//===========N TIMES===============//
+// write a function callNTimes that takes two arguments: times as a number, and fun as a function. It will call that function for that many times. 
+
+fun = "Hello, world!"
```

Comment:
We have a few options. If you want to use a loop we can use a for loop or a while loop. Here's a couple example. 
`function fun() {
    console.log("Hello World!");
}
function callNTimes(times, fun) {
    // If you want to use a loop we can try
    for (let i = 0; i < times; i++) {
        fun();
    }
}
console.log(callNTimes(5, fun)) //does not return the number of times

function fun() {
    console.log("Hello World!");
}
function callNTimes(times, fun) {
    // If you want to use a loop we can try
    let count = 0; 
    while (count < times) {
        fun();
        count++;
    }
}
console.log(callNTimes(5, fun))`
Great work on trying to work your way through this problem. Callbacks are tricky. We can defiantly go over this line by line if any part it seems unclear.